File: sol/solution/0101-0200/0101/Solution.ex
# Definition for a binary tree node.
#
# defmodule TreeNode do
#   @type t :: %__MODULE__{
#           val: integer,
#           left: TreeNode.t() | nil,
#           right: TreeNode.t() | nil
#         }
#   defstruct val: 0, left: nil, right: nil
# end

defmodule Solution do
  def is_symmetric(root), do: is_mirror(root, root)
  defp is_mirror(nil, nil), do: true
  defp is_mirror(nil, _), do: false
  defp is_mirror(_, nil), do: false
  defp is_mirror(%TreeNode{val: v1, left: l1, right: r1}, %TreeNode{val: v2, left: l2, right: r2}) do
    v1 == v2 && is_mirror(l1, r2) && is_mirror(r1, l2)
  end
end

File: sol/solution/0101-0200/0101/Solution.scala
/**
 * Definition for a binary tree node.
 * class TreeNode(_value: Int = 0, _left: TreeNode = null, _right: TreeNode = null) {
 *   var value: Int = _value
 *   var left: TreeNode = _left
 *   var right: TreeNode = _right
 * }
 */
object Solution {
    def isSymmetric(root: TreeNode): Boolean = {
        def isMirror(t1: TreeNode, t2: TreeNode): Boolean = {
            if (t1 == null && t2 == null) true
            else if (t1 == null || t2 == null) false
            else t1.value == t2.value && isMirror(t1.left, t2.right) && isMirror(t1.right, t2.left)
        }
        isMirror(root, root)
    }
}
